### PR TITLE
Meal Sharing app (React1 week3)

### PR DESCRIPTION
**Pull request by katayouny**

### CODE REVIEW

File: vite-project/src/api-url.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+//an environment variable
+const api_host="http://localhost:5000"; 
+
+function api_url(route){
```
Comment (NVedsted) [inline vite-project/src/api-url.js]: Nice idea to extract this type of logic/constants into a file of its own. 👍 You could also consider moving the whole fetching logic into a file like this. 🙂

File: vite-project/src/components/mealsList.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import ListItemWithBorder from "./listItemWithBorder.jsx";
+import api_url from "../api-url.js";
+
+function MealsList() {
+  const [meals, setMeals] = useState([]);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    fetch(api_url("/api/meals")) //calling api function to get the whole route
+      .then((response) => response.json())
```
Comment (NVedsted) [inline vite-project/src/components/mealsList.jsx]: Great use of `.then` method chaining. 😄 You do not have to make this change, but can you figure out how you could use an async function here instead? 🙂

**Review by NVedsted — APPROVED** [review]
Great work! 💪 I had some issues getting your back-end running as it was not too happy with a `GROUP BY`, but my database schema was mostly me guessing. 😄 It would be nice to have a SQL script in your code that can generate the database.

I got past it by mocking some data on the back-end and was met with a nice and simple UI perfectly rendering my meals, good job! 🌟 

I have left some reflection questions; feel free to reach out if you have any questions yourself. 🙂

File: vite-project/src/api-url.js (hunk 1 above)
Comment (katayouny) [reply vite-project/src/api-url.js]: 🙏

File: vite-project/src/components/mealsList.jsx (hunk 2 above)
Comment (katayouny) [reply vite-project/src/components/mealsList.jsx]: I will try to make it. Thank you 🙂

**Comment by katayouny** [comment]
Hey!
Thank you so much.

Could you please explain me about this part? Thank you 😊

'I had some issues getting your back-end running as it was not too happy with a GROUP BY, but my database schema was mostly me guessing. 😄 It would be nice to have a SQL script in your code that can generate the database.'

Unfortunately, I can't get it, as I have just fetched some meals and have not used GROUP BY😊. Maybe you mean, reservations.js? It was part of the previous homework, but I wanted to have it also here 🙂

**Comment by NVedsted** [comment]
Hey, yes, I will try my best to elaborate. 😄 I only spent a brief amount of time trying to get the back-end working as I wanted to focus on the React-part of course. I think it was line 56 in `meals.js` that caused an issue, but it might very well be my database that was improperly setup since I just guessed what tables had to be present, haha. 🙂

**Comment by katayouny** [comment]
Hey Nicklas,

Thank you so much for the explanation 😊.
I have a plan to replace this snippet of code by a more simple one. I hope
I can do it in a near future.😊
Thanks again.🙏

**Pull request closed without merging**